File: 2788-split-strings-by-separator/2788-split-strings-by-separator.js
/**
 * @param {string[]} words
 * @param {character} separator
 * @return {string[]}
 */
var splitWordsBySeparator = function(words, separator) {
    const result = [];
    for(word of words){
        for(number of word.split(separator)){
            if(number) result.push(number);
        }
    }
    return result;
};

File: 2788-split-strings-by-separator/2788-split-strings-by-separator.java
class Solution {
    public List<String> splitWordsBySeparator(List<String> words, char separator) {
        ArrayList<String> numbers = new ArrayList<String>();
        for(String s: words){
            String[] temp = s.split("["+ separator + "]");
            for(String number: temp){
                if(!number.equals("")) numbers.add(number);
            }
        }
        return numbers;
    }
}